include/exadg/poisson/spatial_discretization/operator.cpp

```diff
@@ -501,7 +501,8 @@ Operator<dim, n_components, Number>::solve(VectorType &       sol,
   if(param.spatial_discretization == SpatialDiscretization::CG)
   {
     laplace_operator.set_time(time);
-    laplace_operator.set_inhomogeneous_boundary_values(sol);
+    laplace_operator.set_inhomogeneous_constrained_values(
+      sol, false /* periodicity_and_hanging_node_constraints_only */);
   }
 
```

Comment:
I think from this code here it is clear that we enforce constraints again because they are zero otherwise in the CG case. I think this is rooted in the DG case being the default in ExaDG and we try to be compatible with that using a CG approach. Hence, I think that considering for hanging node and periodicity constraints here as well is not too far off.